### ISSUE TITLE
Add side support for [[files]]

### ISSUE THREAD

**Issue by Stealthii** [packwiz/packwiz#82]
Currently any file defined under the files section is downloaded by packwiz, regardless of server or client.

It would be nice to support side in this, to allow resource pack files to solely be downloaded by client runs, or server configuration files that only the server needs.

**Comment by comp500** [comment]
My plan for this is really the opposite; I want to get rid of index metadata as much as possible for various reasons (code complexity, hash conflicts, version control churn). But I do want to add folder-specific metadata that would allow this (see #78) and allow .toml metadata files to reference local files.

**Comment by rhullah** [comment]
Hate to bring up an old issue, but I came across this issue (again) today. Is there still no way to specify a `side` for assets w/in the packwiz folder?  Is a `side` only supported in the `*.pw.toml` files?  Is there any workaround to achieve this?

**Comment by matteoturini** [comment]
It would be good to split up mods that only work on client/server. For example, I manage a modpack and I would like to add the client optimization mods directly to it, but if that means having them download automatically on the server I would need to distribute them separately, maintain 2 packwiz modpacks, or delete them manually on the server. All of which are way more work than should happen.

**Comment by matteoturini** [comment]
One way to make this not so specialized is to make groups of mods, like ``--group`` flag on the installation, and specify this on installation/bootstrap.

For example:

On setup:
```bash
packwiz mr install Sodium --group client
packwiz mr install Create --group server
```

On the server:
```
 java -jar packwiz-installer-bootstrap.jar -g --group server https://[your-server]/pack.toml
```

On the client:
```
"$INST_JAVA" -jar packwiz-installer-bootstrap.jar --group server,client https://[your-server]/pack.toml
```

**Comment by matteoturini** [comment]
Well the docs certainly don't give that information.  Does the curse forge command also do this?

**Comment by comp500** [comment]
There is a bit of documentation in [the packwiz-installer tutorial](https://packwiz.infra.link/tutorials/installing/packwiz-installer/#using-a-modpack-with-a-server) (and [the reference docs](https://packwiz.infra.link/reference/pack-format/mod-toml/)) but there should definitely be a page specifically for sided/optional mods. The curseforge commands don't do this automatically since CurseForge doesn't provide this information, but you can set the value of `side` yourself.

**Comment by matteoturini** [comment]
What are the possible values of side? `client` and `both`?

**Comment by comp500** [comment]
`both`, `client` and `server` as stated in [the mod.pw.toml reference](https://packwiz.infra.link/reference/pack-format/mod-toml/).

**Comment by matteoturini** [comment]
A command to add these files would be nice

**Comment by rhullah** [comment]
> You can do this by creating a `<file_name>.pw.toml` in the appropriate directory, pointing it to a URL hosting the file you wish to include, and changing its `side` value.

Yeah, I did know about that as I've done that myself.  It's kinda sad that you have to go through that much work to use the `side` capability when packwiz already downloads files "local" to the pack.

**Comment by WxAaRoNxW** [comment]
Would it be difficult to add a feature to add a field `side` in index.toml?
